### PR TITLE
feat: Add possibility to discover archived projects with `catalog-backend-module-gitlab`

### PR DESCRIPTION
**Pull request by mkaliszewski**
## Hey, I just made a Pull Request!

Based on issue I created earlier: https://github.com/backstage/backstage/issues/27184

In this PR I added another option to `catalog-backend-module-gitlab` config called `includeArchivedRepos` (with default false value).

If that parameter is set to `true` in `app-config.yaml`, then the discovery phase includes archived projects.

Example usage in `app-config.yaml`:

```
catalog:
  ...
  providers:
    gitlab:
      projects:
        host: gitlab.xxx.com
        branch: main
        group: projects
        includeArchivedRepos: true
        schedule:
          frequency: { hours: 12 }
          timeout: { minutes: 20 }

```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — APPROVED** [review]

**Review by jhaals — APPROVED** [review]
Looking good, thanks!

File: .changeset/silver-kings-sort.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-gitlab': minor
+---
+
+Extended the configuration with the includeArchivedRepos property, which allows including repositories when the project is archived.
```
Comment (jhaals) [inline .changeset/silver-kings-sort.md:5]:
```suggestion
Extended the configuration with the `includeArchivedRepos` property, which allows including repositories when the project is archived.
```

**Review by jhaals — APPROVED** [review]

File: .changeset/silver-kings-sort.md (hunk 1 above)
Comment (mkaliszewski) [reply .changeset/silver-kings-sort.md]: Updated, thanks for pointing.

**Comment by mkaliszewski** [comment]
> Just a minor thing in the changeset which should hopefully make CI green. It did not pass for some reason

@jhaals CI passed 🎉 but I think I don't have an access to write, thus merge the PR

**Review by jhaals — APPROVED** [review]

**Pull request merged**